src/manager/github/github_pr_request.rs

```diff
@@ -0,0 +1,751 @@
+//! Defines a struct for handling requests from github to start two runs of the same test on the
+//! base and head commits for a PR
+
+use crate::manager::github::{util, Error};
+use crate::manager::notification_handler::NotificationHandler;
+use crate::manager::test_runner::TestRunner;
+use crate::models::run_group::RunGroupData;
+use crate::models::run_group_is_from_github::{NewRunGroupIsFromGithub, RunGroupIsFromGithubData};
+use crate::models::test::TestData;
+use diesel::PgConnection;
+use log::{debug, error};
+use serde::Deserialize;
+use serde_json::Value;
+use uuid::Uuid;
+
+/// Represents the data received from a GitHub Actions request to start two runs of the same test on
+/// separate branches (the PR use case)
+///
+/// `test_input_key` and `eval_input_key` each respectively refer to the key in the test_input and
+/// eval_input for the test that should be filled with a build generated by CARROT using the
+/// specified `software_name` and commits.  `author` refers to the Github username of the person
+/// who triggered the request in GitHub by creating a comment in the format to trigger a test run
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct GithubPrRequest {
+    pub test_name: String,
+    pub test_input_key: Option<String>,
+    pub eval_input_key: Option<String>,
+    pub software_name: String,
+    pub head_commit: String,
+    pub base_commit: String,
+    pub owner: String,
+    pub repo: String,
+    pub issue_number: i32,
+    pub author: String,
+}
+
+impl GithubPrRequest {
```

Comment:
I do think it would be, and I went down a fairly deep rabbit hole of trying to implement this that way.  The problem I ran into is that rust still doesn't support traits with async functions, and we need that in order to really use a trait here the way it would be most helpful.  There is a crate for async traits that I had looked into before and looked into again here, but it is finicky in a way that made it not quite usable for what I was trying to do.  They are actively working on adding async trait support, so hopefully that gets added in a future rust release and this code can be refactored to work that way (ideally if/when we add more request types).

Reply:
Ah.  OK, cool.  That makes sense.  Can you add a todo in the code and / or an issue to track this refactor for later?